teos/src/responder.rs

```diff
@@ -246,88 +195,53 @@ impl Responder {
     ) {
         let tracker = TransactionTracker::new(breach, user_id, status);
 
-        self.trackers
+        if self
+            .dbm
             .lock()
             .unwrap()
-            .insert(uuid, tracker.get_summary());
-
-        let mut tx_tracker_map = self.tx_tracker_map.lock().unwrap();
-        if let Some(map) = tx_tracker_map.get_mut(&tracker.penalty_tx.txid()) {
-            map.insert(uuid);
+            .store_tracker(uuid, &tracker)
+            .is_ok()
+        {
+            log::info!("New tracker added (uuid={uuid})");
         } else {
-            tx_tracker_map.insert(tracker.penalty_tx.txid(), HashSet::from_iter(vec![uuid]));
+            log::error!(
+                "Failed to store tracker in database (uuid={uuid}). It might be already stored."
+            );
```

Comment:
I think that's the reason.

Reply:
Or might be that I didn't want to 1- do the check, 2- release the DB lock before getting into another function `add_tracker`, 3- then insert the tracker there.
Because generally when the lock is release you are not sure anymore that the DB is still in the same state you last seen it.